core/src/mender-client.c

```diff
@@ -1305,9 +1358,12 @@ mender_client_update_work_function(void) {
                         if (MENDER_OK == (ret = mender_check_artifact_requirements(mender_artifact_ctx, deployment))) {
 #ifdef CONFIG_MENDER_PROVIDES_DEPENDS
                             /* Add the new provides to the deployment data (we need the artifact context) */
-                            char *new_provides = NULL;
-                            if (MENDER_OK == (ret = mender_prepare_new_provides(mender_artifact_ctx, &new_provides))) {
+                            char       *new_provides  = NULL;
+                            const char *artifact_name = NULL;
+                            if (MENDER_OK == (ret = mender_prepare_new_provides(mender_artifact_ctx, &new_provides, &artifact_name))) {
                                 cJSON_AddStringToObject(mender_client_deployment_data, "provides", new_provides);
+                                /* Replace artifact_name with the one from provides */
+                                cJSON_ReplaceItemInObject(mender_client_deployment_data, "artifact_name", cJSON_CreateString(artifact_name));
```

Comment:
Why? This seems wrong to me. The whole point of having two is that an Artifact can in theory have a name and also _provide_ an artifact_name.

(I guess I relates to my previous comment - we miss-understood a bit each other here)